ts/src/main/scala/ch/epfl/ts/component/fetch/HistDataCSVFetcher.scala

```diff
@@ -79,25 +92,45 @@ class HistDataCSVFetcher(dataDir: String, currencyPair: String,
   val timer = new Timer()
   timer.schedule(new SendQuotes, 0)
   class SendQuotes extends java.util.TimerTask {
+    
     def run() {
-      if (quoteIndex < allQuotes.length) {
-        // Update the iterator variables
-        var currQ = allQuotes( quoteIndex )
-        var nextQ = allQuotes( List(quoteIndex + 1, allQuotes.length).min )
-        quoteIndex = quoteIndex + 1;
-        
-        // Send the quote and schedule next call
-        callback(currQ)
+      // Get the current quote and send it
+      callback(currQ)
+  
+      if (allQuotes.hasNext) {
+        // If there is a next quote, schedule the next call
+        currQ = nextQ
+        nextQ = allQuotes.next
         timer.schedule(new SendQuotes(), (1 / speed * (nextQ.timestamp - currQ.timestamp)).toInt)
       } else {
-        timer.cancel();
+        // Nothing more to do here, boys!
+        timer.cancel() 
       }
     }
+    
   }
   
-  // TODO
+  /**
+   * Saves the quotes this fetcher has fetched (i.e. quotes that are stored in val allQuotes)
+   * in an sqlite database of a given name.
+   * 
+   * @param   filename    Name of the DB file the quotes will be saved to, the final file
+   *                      will be called <filename>.db
+   */
   def loadInPersistor(filename: String) {
-    throw new UnsupportedOperationException("loadInPersistor is not yet implemented.");
+    val persistor = new QuotePersistor(filename)
+    while(allQuotes.hasNext) {
+      println("Loading 100000 records into persistor...")
+      var quoteBatch = List[Quote]()
+      var batchsize = 0
+      while(batchsize < 100000 && allQuotes.hasNext) {
```

Comment:
Refactor the number to a `val` (to have consistency between the println and the actual loop).